src/JsonSchema/BackwardCompatibleSchemaFactory.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\JsonSchema;
+
+use ApiPlatform\Metadata\Operation;
+
+/**
+ * This factory decorates range integer and number properties to keep Draft 4 backward compatibility.
+ *
+ * @see https://github.com/api-platform/core/issues/6041
+ *
+ * @internal
+ */
+final class BackwardCompatibleSchemaFactory implements SchemaFactoryInterface, SchemaFactoryAwareInterface
+{
+    public const SCHEMA_DRAFT4_VERSION = 'draft_4';
+
+    public function __construct(private readonly SchemaFactoryInterface $decorated)
+    {
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function buildSchema(string $className, string $format = 'json', string $type = Schema::TYPE_OUTPUT, Operation $operation = null, Schema $schema = null, array $serializerContext = null, bool $forceCollection = false): Schema
+    {
+        $schema = $this->decorated->buildSchema($className, $format, $type, $operation, $schema, $serializerContext, $forceCollection);
+
+        if (!isset($serializerContext[self::SCHEMA_DRAFT4_VERSION])) {
```

Comment:
```suggestion
        if ($serializerContext[self::SCHEMA_DRAFT4_VERSION] ?? false) {
```

Reply:
will fix your issue

Reply:
I will add this, but the error is because I am doing `null + []` [example](https://3v4l.org/JjNJ0) in the assertion function